### ISSUE TITLE
template destination backwards compatibility 

### ISSUE THREAD

**Issue by tgross** [hashicorp/nomad#9148]
Prior to Nomad 0.12.5, you could use `${NOMAD_SECRETS_DIR}/mysecret.txt` as the `template.destination`. With the recent security patch (https://github.com/hashicorp/nomad/pull/9139) that will get interpolated as the absolute path `/secrets/mysecret.txt` and get treated as though it were a sandbox escape. Note the leading `/`!

The relative path `secrets/mysecret.txt` works just fine as expected. And documentation is pretty clear that the destination path is supposed to be _relative_ to the working directory, but we do in fact allow absolute paths so long as they don't escape the file sandbox (so you could do `/var/nomad/allocs/:alloc_id/:task_name/secrets/mysecret.txt` if you were so inclined). But this is pretty lousy UX and a nasty surprise for folks.

At the very least, this change should be documented.

(Noticed while working on https://github.com/hashicorp/nomad/issues/9144 https://github.com/hashicorp/nomad/pull/9146)

**Comment by tgross** [comment]
I'm realizing where the issue is: the tricky bit with this is that fixing it without somehow special casing `NOMAD_SECRETS_DIR`, etc. would make the behavior of the `disable_file_sandbox` ambiguous. Maybe it's worth special-casing for those [task directory](https://www.nomadproject.io/docs/runtime/environment#task-directories) env vars.

**Comment by tgross** [comment]
After an internal discussion w/ @schmichael and @krishicks we've realized that this breaks any case where someone was passing an absolute directory path in the `destination` field. Although the docs say "relative-to", it wasn't enforced at all so of course we've baked in behaviors that depend on it.

The fix will be to revert the behavior around the `destination` field to unconditionally prepend the task working directory, and _then_ do the escape check.
